common/gce.py

```diff
@@ -31,60 +29,14 @@ def initialize():
-    request = zone_operations.list(project=project,
-                                   zone=zone,
-                                   orderBy='creationTimestamp desc')
+def get_terminated_instances(project, zone):
+    """Returns a list of terminated instances in |project| and |zone|."""
+    instances = thread_local.service.instances()
+    request = instances.list(project=project, zone=zone)
     while request is not None:
         response = request.execute()
-        for operation in response['items']:
-            yield operation
-
-        request = zone_operations.list_next(previous_request=request,
-                                            previous_response=response)
-
-
-def get_preempted_operations(operations):
-    """Generator that yields GCE preempted operations in |operations|."""
-    # This endpoint doesn't support filtering by time (despite implications
-    # otherwise). Instead it supports ordering by time. It also supports
-    # filtering by operation but doesn't support it when combined with ordering.
-    # So we must filter manually.
-    # See the link below for an example of this query in action.
-    # https://cloud.google.com/compute/docs/reference/rest/v1/zoneOperations/list?apix_params=%7B%22project%22%3A%22fuzzbench%22%2C%22zone%22%3A%22us-central1-a%22%2C%22filter%22%3A%22(operationType%20%3D%20%5C%22compute.instances.preempted%5C%22)%22%2C%22orderBy%22%3A%22creationTimestamp%20desc%22%7D
-    for operation in operations:
-        if operation['operationType'] == 'compute.instances.preempted':
-            yield operation
-
-
-def filter_by_end_time(min_end_time, operations):
-    """Generator that yields GCE operations in |operations| that finished before
-    |min_end_time|. |operations| must be an iterable that is ordered by time."""
-    for operation in operations:
-        end_time = operation.get('endTime')
-        if not end_time:
-            # Try to handle cases where the operation hasn't finished.
-            yield operation
-            continue
-
-        operation_end_time = dateutil.parser.isoparse(end_time)
-        if operation_end_time < min_end_time:
-            break
-        yield operation
-
-
-def get_base_target_link(experiment_config):
-    """Returns the base of the target link for this experiment so that
-    get_instance_from_preempted_operation can return the instance."""
-    return ('https://www.googleapis.com/compute/v1/projects/{project}/zones/'
-            '{zone}/instances/').format(
-                project=experiment_config['cloud_project'],
-                zone=experiment_config['cloud_compute_zone'])
-
-
-def get_instance_from_preempted_operation(operation, base_target_link) -> str:
-    """Returns the instance name from a preempted |operation|."""
-    return operation['targetLink'][len(base_target_link):]
+        for instance in response['items']:
+            if instance['status'] == 'TERMINATED':
+                yield instance['name']
```

Comment:
nit: no reason to yield since you are converting to list anyway in one call site.